packages/components/src/avatar/avatar.tsx

```diff
@@ -1,127 +1,382 @@
+  backgroundColor?: ColorTokens
+}
+
+type WalletAvatarProps = {
+  type: 'wallet'
+  size: 80 | 48 | 32 | 28 | 20
+  name: string
+  backgroundColor?: ColorTokens
+}
+
+type ChannelAvatarProps = {
+  type: 'channel'
+  size: 80 | 32 | 24 | 20
+  emoji: string
+  backgroundColor?: ColorTokens
+  background?: ColorTokens
+  lock?: 'locked' | 'unlocked'
+}
+
+type CommunityAvatarProps = {
+  type: 'community'
+  size: 80 | 32 | 24 | 20
+  name: string
+  src?: string
+  backgroundColor?: ColorTokens
+}
+
+type AccountAvatarProps = {
+  type: 'account'
+  size: 80 | 48 | 32 | 28 | 24 | 20
+  name: string
+  src?: string
+  backgroundColor?: ColorTokens
 }
 
+type IconAvatarProps = {
+  type: 'icon'
+  size: 48 | 32 | 20
+  icon: React.ReactElement
+  backgroundColor?: ColorTokens
+  color?: ColorTokens
+}
+
+type AvatarProps =
+  | UserAvatarProps
+  | GroupAvatarProps
+  | WalletAvatarProps
+  | ChannelAvatarProps
+  | CommunityAvatarProps
+  | AccountAvatarProps
+  | IconAvatarProps
+
 type ImageLoadingStatus = 'idle' | 'loading' | 'loaded' | 'error'
 
-const Avatar = (props: Props) => {
-  const {
-    src,
-    size,
-    shape = 'circle',
-    outline = false,
-    indicator = 'none',
-  } = props
+const userPaddingSizes: Record<UserAvatarProps['size'], number> = {
+  '80': 4,
+  '56': 2,
+  '48': 2,
+  '32': 2,
+  '28': 0,
+  '24': 0,
+  '20': 0,
+  '16': 0,
+}
+
+const accountRadiusSizes: Record<AccountAvatarProps['size'], RadiusTokens> = {
+  '80': '$16',
+  '48': '$12',
+  '32': '$10',
+  '28': '$8',
+  '24': '$8',
+  '20': '$6',
+}
+
+const channelEmojiSizes: Record<ChannelAvatarProps['size'], TextProps['size']> =
+  {
+    // todo: design review
+    '80': 27,
+    '32': 15,
+    '24': 13,
+    '20': 11,
+  }
+
+const textSizes: Record<NonNullable<AvatarProps['size']>, TextProps['size']> = {
+  '80': 27,
+  '56': 19,
+  '48': 19,
+  '32': 15,
+  '28': 13,
+  '24': 13,
+  '20': 11,
+  '16': 11,
+}
+
+const groupMembersIconSizes: Record<
+  GroupAvatarProps['size'],
+  IconProps['size'] | number // to scales SVG
+> = {
+  // todo: design review
+  '80': 36,
+  '48': 20,
+  '32': 16,
+  '28': 16,
+  '20': 12,
+}
+
+const channelLockIconVariants: Record<
+  ChannelAvatarProps['size'],
+  {
+    baseVariant: GetStyledVariants<typeof LockBase>['variant']
+    iconSize: IconProps['size'] | number // to scales SVG
+  }
+> = {
+  // todo: design review
+  '80': { baseVariant: 80, iconSize: 40 },
+  '32': { baseVariant: 24, iconSize: 12 },
+  '24': { baseVariant: 24, iconSize: 12 },
+  '20': { baseVariant: 20, iconSize: 12 },
+}
+
+const Avatar = (props: AvatarProps) => {
+  const colorHash = 'colorHash' in props ? props.colorHash : undefined
+  const identiconRing = useMemo(() => {
+    if (colorHash) {
+      const gradient = generateIdenticonRing(colorHash)
+      return `conic-gradient(from 90deg, ${gradient})`
+    }
+  }, [colorHash])
 
   const [status, setStatus] = useState<ImageLoadingStatus>('idle')
+  const imageBackgroundColor =
```

Comment:
fixme!

Reply:
else

<img width="318" alt="image" src="https://user-images.githubusercontent.com/13265126/232971133-229d6cab-44bf-4a70-bd27-596159329030.png">
dark overlap
<img width="321" alt="image" src="https://user-images.githubusercontent.com/13265126/232971216-3415eda5-00b7-4fe0-9937-30bd7632d581.png">
no overlap

<img width="159" alt="image" src="https://user-images.githubusercontent.com/13265126/232971475-db02a9aa-71aa-4499-96b3-1319ae34637f.png">
dark background (default)
<img width="126" alt="image" src="https://user-images.githubusercontent.com/13265126/232971524-c7abeb04-6735-4b59-8a69-30510ef8db7b.png">
light background (default for channel avatar)

Reply:
Done at https://github.com/status-im/status-web/pull/371/commits/9ffced12a79a6f65855be2582f2d4abc31ff1be8.